pkg/icsp/icsp.go

```diff
@@ -0,0 +1,226 @@
+	glog.V(100).Infof(
+		"Initializing new ICSPBuilder structure with the following params: \nName: %s\n"+
+			"Source: %s \n"+"Mirrors: %v\n", name, source, mirrors)
+
+	icspBuilder := &ICSPBuilder{
+		apiClient: apiClient,
+		Definition: &v1alpha1.ImageContentSourcePolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: v1alpha1.ImageContentSourcePolicySpec{
+				RepositoryDigestMirrors: []v1alpha1.RepositoryDigestMirrors{
+					{
+						Source:  source,
+						Mirrors: mirrors,
+					},
+				},
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the ImageContentSourcePolicy is empty")
+
+		icspBuilder.errorMsg = "ImageContentSourcePolicy 'name' cannot be empty"
+	}
+
+	if source == "" {
+		glog.V(100).Infof("The Source of the ImageContentSourcePolicy is empty")
+
+		icspBuilder.errorMsg = "ImageContentSourcePolicy 'source' cannot be empty"
+	}
+
+	if len(mirrors) == 0 {
+		glog.V(100).Infof("The mirrors of the ImageContentSourcePolicy are empty")
+
+		icspBuilder.errorMsg = "ImageContentSourcePolicy 'mirrors' cannot be empty"
+	}
+
+	return icspBuilder
+}
+
+// Exists check if object exists in the cluster.
+func (builder *ICSPBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if ImageContentSourcePolicy %s exists", builder.Definition.Name)
+
+	var err error
+
+	builder.Object, err = builder.apiClient.ImageContentSourcePoliciesGetter.ImageContentSourcePolicies().Get(
+		context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Pull pulls object definition from cluster to ICSPBuilder struct.
+func Pull(apiClient *clients.Settings, name string) (*ICSPBuilder, error) {
+	glog.V(100).Infof("Pulling existing ICSP name: %s under namespace: %s", name)
+
+	builder := ICSPBuilder{
+		apiClient: apiClient,
+		Definition: &v1alpha1.ImageContentSourcePolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		builder.errorMsg = "ImageContentSourcePolicy 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("ImageContentSourcePOlicy object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Create makes a ImageContentSourcePolicy in the cluster and stores the created object in struct.
+func (builder *ICSPBuilder) Create() (*ICSPBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating namespace %s", builder.Definition.Name)
+
+	var err error
+
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.ImageContentSourcePolicies().Create(
+			context.TODO(), builder.Definition, metav1.CreateOptions{})
+	} else {
+		glog.V(100).Infof("\nImageContentSourcePolicy %s already exists\n", builder.Definition.Name)
```

Comment:
I don't think we need an else condition for the glog

Reply:
Why not if we are doing verbose (100) ?

Reply:
Removed